contributors/design-proposals/troubleshoot-running-pods.md

```diff
@@ -0,0 +1,666 @@
+# Troubleshoot Running Pods
+
+This proposal seeks to add first class support for troubleshooting by creating a
+mechanism to execute a shell or other troubleshooting tools inside a running pod
+without requiring that the associated container images include such tools.
+
+## Motivation
+
+### Development
+
+Many developers of native Kubernetes applications wish to treat Kubernetes as an
+execution platform for custom binaries produced by a build system. These users
+can forgo the scripted OS install of traditional Dockerfiles and instead `COPY`
+the output of their build system into a container image built `FROM scratch`.
+This confers several advantages:
+
+1.  **Minimal images** lower operational burden and reduce attack vectors.
+1.  **Immutable images** improve correctness and reliability.
+1.  **Smaller image size** reduces resource usage and speeds deployments.
+
+The disadvantage of using containers built `FROM scratch` is the lack of system
+binaries provided a base Operation System image makes it difficult to
+troubleshoot running containers. Kubernetes should enable troubleshooting pods
+regardless of the contents of the container image.
+
+### Operations and Support
+
+As Kubernetes gains in popularity, it's becoming the case that a person
+troubleshooting an application is not necessarily the person who built it.
+Operations staff and Support organizations want the ability to attach a "known
+good" or automated debugging environment to a pod.
+
+## Requirements
+
+A solution to troubleshoot arbitrary container images MUST:
+
+*   troubleshoot arbitrary running containers with minimal prior configuration
+*   allow access to shared pod namespaces and the file systems of individual
+    containers
+*   fetch troubleshooting utilities at debug time rather than at the time of pod
```

Comment:
fetch troubleshooting utilities at debug time rather than at the time of pod creation because...
- want to avoid image pulling?
- concerns about pod startup latency?

more detail here would be helpful.